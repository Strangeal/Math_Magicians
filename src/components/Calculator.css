@import url('https://fonts.googleapis.com/css2?family=Lato:ital,wght@0,100;0,300;0,400;0,700;0,900;1,100;1,300;1,400;1,700;1,900&display=swap');

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.calculator {
  width: 40%;
  margin: auto;
  outline: none;
  background-color: #000;
  font-family: 'Lato', sans-serif;
  margin-top: 2rem;
  border: 2px solid #a5a5a5;
}

.calculator__Ui {
  padding: 1.3rem;
}

/* screen */
.screen {
  width: 100%;
  padding: 1rem;
  margin-bottom: 2.5rem;
  border: none;
  background-color: #161515;
  color: #fff;
  font-size: 1.3rem;
  font-weight: 600;
}

.calculator__keys {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-gap: 0.8rem;
}

button {
  background: #333;
  border-radius: 12px;
  border: none;
  padding: 0;
  cursor: pointer;
  outline-offset: 4px;
}

.front {
  padding: 1.9rem 0.5rem;
  display: block;
  border-radius: 12px;
  font-size: 1.25rem;
  background: #333;
  color: #fff;
  transform: translateY(-4px);
  font-weight: 700;
}

button:active .front {
  transform: translateY(-2px);
}

button:nth-of-type(4n) .front,
button:last-child .front {
  background: #fa6400;
}

button:nth-of-type(17) {
  grid-column: 1 / span 2;
}

button:nth-of-type(-n+3) .front {
  background-color: #a5a5a5;
}
