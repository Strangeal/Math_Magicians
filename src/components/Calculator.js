import React, { useState } from 'react';
import calculate from '../logic/calculate';
import '../App.css';

export const Calculator = () => {
  const [counter, setCounter] = useState({
    total: '0',
    next: null,
    operation: null,
  });

  const buttonKeys = [
    'AC', '+/-', '%', 'รท',
    '7', '8', '9', 'x',
    '6', '5', '4', '-',
    '3', '2', '1', '+',
    '0', '.', '=',
  ];

  const onClickHandler = (e) => {
    const {
      target: { textContent: buttonName },
    } = e;

    const results = calculate(counter, buttonName);
    setCounter(results);
  };

  const { total, next } = counter;
  return (

    <div className="calculator__container">
      <h3>Let do some Maths!</h3>
      <div className="calculator">
        <div className="calculator__Ui">
          <div className="screen">{next || total || 0}</div>
          <div className="calculator__keys">
            {buttonKeys.map((each) => (
              <button
                key={each}
                onClick={(e) => onClickHandler(e)}
                type="button"
              >
                <span className="front">{each}</span>
              </button>
            ))}
          </div>
        </div>
      </div>
    </div>
  );
};
export default Calculator;
