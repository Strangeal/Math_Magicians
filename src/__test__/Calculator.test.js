import React from "react";
import { render } from "@testing-library/react";
import Calculator from "../components/Calculator";

it("renders to match the snapshot", () => {
  const tree = render(<Calculator />);
  expect(tree).toMatchSnapshot();
});
