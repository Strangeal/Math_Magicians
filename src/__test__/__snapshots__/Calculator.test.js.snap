// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when there is item 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="calculator__container"
      >
        <h3>
          Let do some Maths!
        </h3>
        <div
          class="calculator"
        >
          <div
            class="calculator__Ui"
          >
            <div
              class="screen"
            >
              0
            </div>
            <div
              class="calculator__keys"
            >
              <button
                type="button"
              >
                <span
                  class="front"
                >
                  AC
                </span>
              </button>
              <button
                type="button"
              >
                <span
                  class="front"
                >
                  +/-
                </span>
              </button>
              <button
                type="button"
              >
                <span
                  class="front"
                >
                  %
                </span>
              </button>
              <button
                type="button"
              >
                <span
                  class="front"
                >
                  รท
                </span>
              </button>
              <button
                type="button"
              >
                <span
                  class="front"
                >
                  7
                </span>
              </button>
              <button
                type="button"
              >
                <span
                  class="front"
                >
                  8
                </span>
              </button>
              <button
                type="button"
              >
                <span
                  class="front"
                >
                  9
                </span>
              </button>
              <button
                type="button"
              >
                <span
                  class="front"
                >
                  x
                </span>
              </button>
              <button
                type="button"
              >
                <span
                  class="front"
                >
                  6
                </span>
              </button>
              <button
                type="button"
              >
                <span
                  class="front"
                >
                  5
                </span>
              </button>
              <button
                type="button"
              >
                <span
                  class="front"
                >
                  4
                </span>
              </button>
              <button
                type="button"
              >
                <span
                  class="front"
                >
                  -
                </span>
              </button>
              <button
                type="button"
              >
                <span
                  class="front"
                >
                  3
                </span>
              </button>
              <button
                type="button"
              >
                <span
                  class="front"
                >
                  2
                </span>
              </button>
              <button
                type="button"
              >
                <span
                  class="front"
                >
                  1
                </span>
              </button>
              <button
                type="button"
              >
                <span
                  class="front"
                >
                  +
                </span>
              </button>
              <button
                type="button"
              >
                <span
                  class="front"
                >
                  0
                </span>
              </button>
              <button
                type="button"
              >
                <span
                  class="front"
                >
                  .
                </span>
              </button>
              <button
                type="button"
              >
                <span
                  class="front"
                >
                  =
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="calculator__container"
    >
      <h3>
        Let do some Maths!
      </h3>
      <div
        class="calculator"
      >
        <div
          class="calculator__Ui"
        >
          <div
            class="screen"
          >
            0
          </div>
          <div
            class="calculator__keys"
          >
            <button
              type="button"
            >
              <span
                class="front"
              >
                AC
              </span>
            </button>
            <button
              type="button"
            >
              <span
                class="front"
              >
                +/-
              </span>
            </button>
            <button
              type="button"
            >
              <span
                class="front"
              >
                %
              </span>
            </button>
            <button
              type="button"
            >
              <span
                class="front"
              >
                รท
              </span>
            </button>
            <button
              type="button"
            >
              <span
                class="front"
              >
                7
              </span>
            </button>
            <button
              type="button"
            >
              <span
                class="front"
              >
                8
              </span>
            </button>
            <button
              type="button"
            >
              <span
                class="front"
              >
                9
              </span>
            </button>
            <button
              type="button"
            >
              <span
                class="front"
              >
                x
              </span>
            </button>
            <button
              type="button"
            >
              <span
                class="front"
              >
                6
              </span>
            </button>
            <button
              type="button"
            >
              <span
                class="front"
              >
                5
              </span>
            </button>
            <button
              type="button"
            >
              <span
                class="front"
              >
                4
              </span>
            </button>
            <button
              type="button"
            >
              <span
                class="front"
              >
                -
              </span>
            </button>
            <button
              type="button"
            >
              <span
                class="front"
              >
                3
              </span>
            </button>
            <button
              type="button"
            >
              <span
                class="front"
              >
                2
              </span>
            </button>
            <button
              type="button"
            >
              <span
                class="front"
              >
                1
              </span>
            </button>
            <button
              type="button"
            >
              <span
                class="front"
              >
                +
              </span>
            </button>
            <button
              type="button"
            >
              <span
                class="front"
              >
                0
              </span>
            </button>
            <button
              type="button"
            >
              <span
                class="front"
              >
                .
              </span>
            </button>
            <button
              type="button"
            >
              <span
                class="front"
              >
                =
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
